drivers/can/CMakeLists.txt

```diff
@@ -20,28 +20,30 @@ zephyr_library_sources_ifdef(CONFIG_CAN_STM32H7_FDCAN can_stm32h7_fdcan.c)
 zephyr_library_sources_ifdef(CONFIG_CAN_TCAN4X5X     can_tcan4x5x.c)
 zephyr_library_sources_ifdef(CONFIG_CAN_RCAR         can_rcar.c)
 zephyr_library_sources_ifdef(CONFIG_CAN_NUMAKER      can_numaker.c)
+zephyr_library_sources_ifdef(CONFIG_CAN_SJA1000      can_sja1000.c)
+zephyr_library_sources_ifdef(CONFIG_CAN_ESP32_TWAI   can_esp32_twai.c)
+zephyr_library_sources_ifdef(CONFIG_CAN_KVASER_PCI   can_kvaser_pci.c)
+
+zephyr_library_sources_ifdef(CONFIG_USERSPACE        can_handlers.c)
+zephyr_library_sources_ifdef(CONFIG_CAN_SHELL        can_shell.c)
+zephyr_library_sources_ifdef(CONFIG_CAN_NXP_S32_CANXL    can_nxp_s32_canxl.c)
 
 if(CONFIG_CAN_NATIVE_POSIX_LINUX)
   if(${CMAKE_HOST_SYSTEM_NAME} STREQUAL Linux)
     zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/l2)
-    zephyr_library_compile_definitions(NO_POSIX_CHEATS)
-    zephyr_library_compile_definitions(_BSD_SOURCE)
-    zephyr_library_compile_definitions(_DEFAULT_SOURCE)
-    zephyr_library_sources(
-      can_native_posix_linux.c
-      can_native_posix_linux_socketcan.c
-    )
+    zephyr_library_sources(can_native_posix_linux.c)
+    set_source_files_properties(can_native_posix_linux_socketcan.c
```

Comment:
Note this will only affect the native_posix build of this file (i.e. when this file is build with the rest of Zephyr).
Though as things build fine now for native_sim (and also for native_posix if these are removed), I guess the `_BSD_SOURCE` & `_DEFAULT_SOURCE` are not really needed?
Anyhow, maybe you want to move this inside the `if (CONFIG_NATIVE_APPLICATION)`

(Sidenote: These are the default options used in native_sim `-D_POSIX_C_SOURCE=200809 -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED`)

Reply:
Perhaps they are indeed not needed, but the Linux headers are a nightmare in this regard. I've left them in, but moved them to inside the `if` clause. Thanks.